tests/pkg/expectations-builders/formation_assignment.go

```diff
@@ -0,0 +1,109 @@
+package expectations_builders
+
+import (
+	"github.com/kyma-incubator/compass/tests/pkg/fixtures"
+	"k8s.io/utils/strings/slices"
+)
+
+type FAExpectationsBuilder struct {
+	expectations map[string]map[string]fixtures.AssignmentState
+}
+
+func NewFAExpectationsBuilder() *FAExpectationsBuilder {
+	return &FAExpectationsBuilder{
+		expectations: map[string]map[string]fixtures.AssignmentState{},
+	}
+}
+
+func (b *FAExpectationsBuilder) GetExpectations() map[string]map[string]fixtures.AssignmentState {
+	return b.expectations
+}
+
+func (b *FAExpectationsBuilder) GetExpectedAssignmentsCount() int {
+	count := 0
+	for _, val := range b.expectations {
+		count += len(val)
+	}
+	return count
+}
+
+func (b *FAExpectationsBuilder) getCurrentParticipants() []string {
+	participants := make([]string, 0, len(b.expectations))
+	for participantID, _ := range b.expectations {
+		participants = append(participants, participantID)
+	}
+
+	return participants
+}
+
+func (b *FAExpectationsBuilder) WithParticipant(participantID string) *FAExpectationsBuilder {
+	if _, ok := b.expectations[participantID]; ok {
+		return b
+	}
+
+	for _, val := range b.expectations {
+		val[participantID] = fixtures.AssignmentState{State: "READY", Config: nil, Value: nil, Error: nil}
+	}
+
+	currentParticipants := b.getCurrentParticipants()
+	b.expectations[participantID] = make(map[string]fixtures.AssignmentState, len(currentParticipants)+1)
+	b.expectations[participantID][participantID] = fixtures.AssignmentState{State: "READY", Config: nil, Value: nil, Error: nil}
+	for _, participant := range currentParticipants {
+		b.expectations[participantID][participant] = fixtures.AssignmentState{State: "READY", Config: nil, Value: nil, Error: nil}
+	}
```

Comment:
The naming is a little bit confusing because we use `participantID` and `participant` for the same type of data(string/ID) but the second seems to be another thing? "getParicipants" func actually returns participant IDs(check the other comment). I think it will be better if we specify the participants by "source" and "target" or another method of differentiation? I'm suggesting this because in the current setup of the tests we have a similar map that is based on map[sourceID]map[targetID]SomeData, or at least in my mind.

e.g.
```
func (b *FAExpectationsBuilder) getCurrentParticipantIDs() []string {
	participantIDs := make([]string, 0, len(b.expectations))
	for participantID, _ := range b.expectations {
		participantIDs = append(participantIDs, participantID)
	}

	return participantIDs
}

func (b *FAExpectationsBuilder) WithParticipant(sourceParticipantID string) *FAExpectationsBuilder {
	if _, ok := b.expectations[sourceParticipantID]; ok {
		return b
	}

	for _, val := range b.expectations {
		val[sourceParticipantID] = fixtures.AssignmentState{State: "READY", Config: nil, Value: nil, Error: nil}
	}

	currentParticipantIDs := b.getCurrentParticipantIDs()
	b.expectations[sourceParticipantID] = make(map[string]fixtures.AssignmentState, len(currentParticipantIDs)+1)
	b.expectations[sourceParticipantID][sourceParticipantID] = fixtures.AssignmentState{State: "READY", Config: nil, Value: nil, Error: nil}
	for _, targetParticipantID := range currentParticipantIDs {
		b.expectations[sourceParticipantID][targetParticipantID] = fixtures.AssignmentState{State: "READY", Config: nil, Value: nil, Error: nil}
	}

	return b
}
```

Reply:
renamed the variables to make it more readable:
participantID -> newParticipantID
val -> expectationsForPreviouslyAddedParticipantTarget
participant - > previouslyAddedParticipantID 
added comments as well as the names become long and may still cause confusion